tests/sftp/hooks/test_sftp.py

```diff
@@ -0,0 +1,84 @@
+from unittest.mock import patch
+
+import pytest
+from airflow.exceptions import AirflowException
+from asyncssh import SFTPNoSuchFile
+
+from astronomer.providers.sftp.hooks.sftp import SFTPHookAsync
+
+
+class MockSFTPClient:
+    def __init__(self):
+        pass
+
+    async def listdir(self, path: str):
+        if path == "/path/does_not/exist/":
+            raise SFTPNoSuchFile("File does not exist")
+        else:
+            return ["..", ".", "file"]
+
+
+class MockSSHClient:
+    def __init__(self):
+        pass
+
+    async def start_sftp_client(self):
+        return MockSFTPClient()
+
+
+@patch("astronomer.providers.sftp.hooks.sftp.SFTPHookAsync._get_conn")
+@pytest.mark.asyncio
+async def test_list_directory_path_does_not_exist(mock_hook_get_conn):
```

Comment:
can you please convert these tests to class-based tests. you can take a look at either core or cncf tests https://github.com/astronomer/astronomer-providers/blob/main/tests/core/sensors/test_external_task.py